File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        int min = nums[0];
        int l = 0;
        int r = nums.length -1;
        while(l < r){
            int middle = l + (r - l)/2;
            if(nums[l] < min){
                min = nums[l];
            }
            if(nums[r] < min){
                min = nums[r];
            }
            if(nums[middle] < min){
                l = l + 1;
            } else{
                r = r - 1;
            }
        }
        return min;        
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        if not nums:
            return []
        prefix = [1] * len(nums) 
        for i in range(1,len(nums)):
            prefix[i] = prefix[i-1] * nums[i-1]
        suffix = [1] * len(nums)
        for i in range(len(nums)-1,0,-1):
            suffix[i-1] = suffix[i] * nums[i]

        res = [1] * len(nums)
        for i in range(len(nums)):
            res[i] = prefix[i] * suffix[i]
        return res


        